lib/internal/Magento/Framework/Data/Collection.php

```diff
@@ -19,8 +17,10 @@
  *
  * @api
  * @since 100.0.2
+ *
+ * @SuppressWarnings(PHPMD.ExcessivePublicCount)
  */
-class Collection implements \IteratorAggregate, \Countable, ArrayInterface, CollectionDataSourceInterface
+class Collection implements \IteratorAggregate, \Countable, OptionSourceInterface, CollectionDataSourceInterface
```

Comment:
Also not clear why you changed interface here

Reply:
ArrayInterface is deprecated.
see https://github.com/magento/magento2/blob/2.4-develop/lib/internal/Magento/Framework/Option/ArrayInterface.php

Reply:
Seems like it’s not really related to the name of this PR, also this isn’t backward compatible change. Can we extract it to separate PR?

Reply:
Ah, i see that svc fails anyway, so we’ll need approval. No need to do anything